app/flatpak-cli-transaction.c

```diff
@@ -501,35 +501,44 @@ operation_error (FlatpakTransaction            *transaction,
   return TRUE; /* Continue */
 }
 
-static void
-end_of_lifed (FlatpakTransaction *transaction,
-              const char         *ref,
-              const char         *reason,
-              const char         *rebase)
+static gboolean
+end_of_lifed_pre (FlatpakTransaction *transaction,
+                  const char         *remote,
+                  const char         *ref,
+                  const char         *reason,
+                  const char         *rebase)
 {
   FlatpakCliTransaction *self = FLATPAK_CLI_TRANSACTION (transaction);
   g_autoptr(FlatpakRef) rref = flatpak_ref_parse (ref, NULL);
-  g_autofree char *msg = NULL;
 
   if (rebase)
-    msg = g_strdup_printf (_("Info: %s is end-of-life, in preference of %s"),
-                           flatpak_ref_get_name (rref), rebase);
+    g_print (_("Info: %s is end-of-life, in preference of %s\n"), flatpak_ref_get_name (rref), rebase);
   else if (reason)
-    msg = g_strdup_printf (_("Info: %s is end-of-life, with reason: %s\n"),
-                           flatpak_ref_get_name (rref), reason);
+    g_print (_("Info: %s is end-of-life, with reason: %s\n"), flatpak_ref_get_name (rref), reason);
 
-  if (flatpak_fancy_output ())
+  if (rebase && remote && !self->disable_interaction)
     {
-      flatpak_table_printer_set_cell (self->printer, self->progress_row, 0, msg);
-      self->progress_row++;
-      flatpak_table_printer_add_span (self->printer, "");
-      flatpak_table_printer_finish_row (self->printer);
-      redraw (self);
+      if (flatpak_yes_no_prompt (FALSE, _("Replace it with %s?"), flatpak_ref_get_name (rref)))
+        {
+          g_autoptr(GError) error = NULL;
+          const char *previous_ids[] = { flatpak_ref_get_name (rref), NULL };
```

Comment:
Don't you need to also read the other previous ids from the deploy? Maybe we can pass those as an argument to the signal?

Reply:
I added it as an argument, but we still need to pass in the older ids.